### PR TITLE
Improve docker input plugin

### PR DESCRIPTION
**Pull request by titilambert**
Hello !
This commit adds `docker info` data in returned metrics and improve test coverage to more than 70%

### CODE REVIEW

**Comment by sparrc** [comment]
thanks @titilambert, can you update the README with the new stats?

**Comment by titilambert** [comment]
Done !

**Comment by sparrc** [comment]
the way the metrics are formatted seems a little inconsistent to me, what about like this?

```
- docker
    - n_used_file_descriptors
    - n_cpus
    - n_containers
    - n_images
    - n_goroutines
    - n_listener_events
    - memory_total
    - pool_blocksize
- docker_data
    - available
    - total
    - used
- docker_metadata
    - available
    - total
    - used
```

**Comment by titilambert** [comment]
@sparrc you're right ! Changes done !

**Comment by tripledes** [comment]
@titilambert hey, looking at the diff just spotted a typo:

docker (**memeory**_total)

Nice job btw!

**Comment by titilambert** [comment]
@tripledes Thanks ! Fixed !

File: plugins/inputs/docker/docker.go [hunk 1]
```diff
@@ -81,6 +112,88 @@ func (d *Docker) Gather(acc telegraf.Accumulator) error {
...
+	metadataFields := make(map[string]interface{})
+	now := time.Now()
+	// Get info from docker daemon
+	info, err := d.client.Info()
+	if err != nil {
+		return err
+	}
+
+	// Add metrics
+	acc.AddFields("docker",
```
Comment (sparrc) [inline plugins/inputs/docker/docker.go]: change these to a single call to AddFields():

```go
fields := map[string]interface{}{
  "n_cpus": ...
  "n_containers": ...
}
acc.AddFields("docker, fields, ...)
```

**Comment by titilambert** [comment]
Done ! I didn't merge `memory_total` because there is `unit` tag with.

**Pull request closed without merging**